boule/tests/test_ellipsoid.py

```diff
@@ -325,3 +325,14 @@ def test_normal_gravity_against_somigliana(ellipsoid):
             ellipsoid.normal_gravity(latitude, height=0),
             normal_gravity_surface(latitude, ellipsoid),
         )
+
+
+@pytest.mark.parametrize("ellipsoid", ELLIPSOIDS, ids=ELLIPSOID_NAMES)
+def test_normal_gravity_computed_on_internal_point(ellipsoid):
+    """
+    Check if warn is raised if height is negative
+    """
+    with warnings.catch_warnings(record=True) as warn:
+        latitude = np.linspace(-90, 90, 100)
```

Comment:
We might want to catch the warnings only for the `ellipsoid.normal_gravity` call, just in case the `np.linspace` raises a warning in the future (might never happen though).
```suggestion
    latitude = np.linspace(-90, 90, 100)
    with warnings.catch_warnings(record=True) as warn:
```